docs/endpoint.md

```diff
@@ -5,14 +5,140 @@ The endpoint is one of the components provided by Tangle-accelerator, running on
+    gnupg \
+    bash-completion \
+    xz-utils
 ```
-$ bazel build //endpoint:wp7702
-$ bazel build //endpoint:sim
+
+Install leaf
+
+```shell
+$ curl -sLO https://downloads.sierrawireless.com/tools/leaf/leaf_latest.deb && sudo dpkg -i leaf_latest.deb
 ```
 
-## HTTPS Connection Support
-The endpoint uses http connection as default. The message which sent to tangle-accelerator has been encrypted. So the HTTP connection would not be unsafe. To build with https connection support, add `--define https=enable` option.
+Create a workspace
+
+```shell
+$ mkdir -p workspace
+$ cd workspace
+```
+
+Setup the wp77xx target profile
+
+```shell
+$ leaf setup legato-stable -p swi-wp77_3.0.0
+```
+
+Finally, make the wp77xx endpoint target. Be careful with the directory of tangle-accelerator. It should be located within the workspace directory.
+
+```shell
+$ git clone https://github.com/DLTcollab/tangle-accelerator.git
+$ cd tangle-accelerator
+$ make TARGET=wp77xx legato # build endpoint as wp77xx target
+$ make TESTS=true TARGET=wp77xx legato # build endpoint as wp77xx target in test mode
 ```
-$ bazel build --define https=enable //endpoint:wp7702
+
+### How to build endpoint application for native target
+
+Install required packages:
+
+```shell
+$ sudo apt-get install -y   \
+    autoconf                \
+    automake                \
+    bash                    \
+    bc                      \
+    bison                   \
+    bsdiff                  \
+    build-essential         \
+    chrpath                 \
+    cmake                   \
+    cpio                    \
+    diffstat                \
+    flex                    \
+    gawk                    \
+    gcovr                   \
+    git                     \
+    gperf                   \
+    iputils-ping            \
+    libbz2-dev              \
+    libcurl4-gnutls-dev     \
+    libncurses5-dev         \
+    libncursesw5-dev        \
+    libsdl-dev              \
+    libssl-dev              \
+    libtool                 \
+    libxml2-utils           \
+    ninja-build             \
+    python                  \
+    python-git              \
+    python-jinja2           \
+    python-pkg-resources    \
+    python3                 \
+    texinfo                 \
+    unzip                   \
+    wget                    \
+    zlib1g-dev
+```
+
+Create a workspace
+
+```shell
+$ mkdir -p workspace
+$ cd workspace
+```
+
+Clone a specific version of `legato`. The `19.07.0` is the preferred stable version.
+
+```shell
+$ repo init -u git://github.com/legatoproject/manifest -m legato/releases/19.07.0.xml # specific legato 19.07.0 version
+$ repo sync
+```
+
+Build legato as native target
+
+```shell
+$ cd legato
+$ make localhost
+```
+
+Checkout the shell to legato shell
+
+```shell
+$ source ./framework/tools/scripts/configlegatoenv
+```
+
+Finally, go back to the root directory of tangle-accelerator. Use GNU Make to build the endpoint application.
```

Comment:
Can we avoid using the description "go back to the root directory"? We can use `pushd` and `popd` instead. There are builtin commands in bash.